### PR TITLE
Search Courses

### PR DESCRIPTION
**Pull request by Sashuteak**
US - https://rally1.rallydev.com/#/431372673576d/search?detail=%2Fuserstory%2F626579295315&keywords=POLITE

image
image

### CODE REVIEW

**Review by IhabAlkabary — APPROVED** [review]

File: custom-my-courses-in-other-orgs.js [hunk 1]
```diff
@@ -99,10 +96,12 @@ class CustomMyCoursesInOtherOrgs extends LocalizeLitMixin(LitElement) {
 	}
 
 	render() {
-		if (this.myCoursesInOtherOrgsList.length === 0 && this.searchText === '') {
-			return this.renderEmptyCoursesContainer();
+		if (this.myCoursesInOtherOrgsList.length === 0 && this.searchText === '' && this.filteredOrgId === '') {
```
Comment (josephsaffron) [inline custom-my-courses-in-other-orgs.js:99]: can we use undefined instead of the empty string to denote no value?

Comment (Sashuteak) [reply custom-my-courses-in-other-orgs.js:99]: If I do this, then I will have to do additional checks and assign undefined values

for example line 182 where this.searchText = e.detail.value;

File: custom-my-courses-in-other-orgs.js [hunk 2]
```diff
@@ -116,19 +115,19 @@ class CustomMyCoursesInOtherOrgs extends LocalizeLitMixin(LitElement) {
 						org-defined-id="${this.orgDefinedId}"
 						@d2l-my-organization-filter-change="${this._filterChanged}"
 					></d2l-my-organization-filter>
-					${this.myCoursesInOtherOrgsList.length === 0 && this.searchText !== '' ? this.renderEmptyCoursesContainer() : html``}
+					${needDisplayEmptyContainer ? this.renderEmptyCoursesContainer('serchEmptyResult') : html``}
```
[the remark this answers is not in the document]
Comment (Sashuteak) [reply custom-my-courses-in-other-orgs.js]: done

File: custom-my-courses-in-other-orgs.js (hunk 1 above)
Comment (Sashuteak) [reply custom-my-courses-in-other-orgs.js]: Done

File: custom-my-courses-in-other-orgs.js [hunk 3]
```diff
@@ -115,19 +113,19 @@ class CustomMyCoursesInOtherOrgs extends LocalizeLitMixin(LitElement) {
...
 			</d2l-list>`;
 	}
 
-	renderEmptyCoursesContainer() {
+	renderEmptyCoursesContainer(containerMassage) {
 		return html`
 			<div class="d2l-msg-container">
 				<div class="d2l-msg-container-inner">
-					<div class="d2l-msg-container-text">${this.localize('emptyListMessage')}</div>
+					<div class="d2l-msg-container-text">${this.localize(containerMassage)}</div>
```
Comment (josephsaffron) [inline custom-my-courses-in-other-orgs.js:128]:
```suggestion
					<div class="d2l-msg-container-text">${this.localize(containerMessage)}</div>
```

**Review by josephsaffron — APPROVED** [review]

**Pull request merged**